recipes/godot-cpp/all/conanfile.py

```diff
@@ -0,0 +1,141 @@
+import glob
+import os
+
+from conans import ConanFile, tools
+from conans.errors import ConanInvalidConfiguration
+
+
+class GodotCppConan(ConanFile):
+    name = "godot-cpp"
+    description = "C++ bindings for the Godot script API"
+    license = "MIT"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/godotengine/godot-cpp"
+    topics = ("game-engine", "game-development", "c++")
+    settings = {
+        "arch": ["x86", "x86_64"],
+        "os": ["Windows", "Linux", "Macos"],
+        "compiler": ["gcc", "clang", "Visual Studio"],
+        "build_type": ["Debug", "Release", "MinSizeRel", "RelWithDebInfo"],
+    }
```

Comment:
There is no need to restrict settings (unless we really know it is not working for other combinations). IMHO it is better to leave it free, maybe this same recipe works using other compilers like `intel` or `apple-clang` (probably a bug here).

Reply:
Fair point. IIRC I would need to set the value of `compiler` to `None`, is that correct?

Reply:
I was too focused on compilers. I will remove the other limitations as soon as i get home

Reply:
I would use just:

```
settings = "os", "arch", "compiler", "build_type"
```

It tells Conan that with each different combination of those values it should generate a different package-ID because the binaries generated are different. No need to list explicitly the values, they are stored (and configured) in the `<cache>/settings.yml` file inside the user folder.